### PR TITLE
Updated the hardcoded dependencies

### PR DESCRIPTION
**Pull request by yashsingh74**
The PR fixes the following changes-

- Allow default `run` args `-it` to be overridden by specifying `DOCKER_RUN_ARGS`.
-  Introduce `GO_IMAGE` and `GO_VERSION` to allow the compiliation docker image to be specified.
- Add logic to handle the case where a non `alpine` golang image is used so that the script no longer blindly attempts to `apk` install `bash` and `tar`. Scripts checks for presence of both commands.
- Avoid using dos2unix, use sed

### CODE REVIEW

**Comment by yashsingh74** [comment]
cc: @squeed

**Comment by yashsingh74** [comment]
@squeed can you review this PR as well.

**Comment by yashsingh74** [comment]
cc: @dcbw

**Review by squeed — APPROVED** [review]

File: scripts/release.sh [hunk 1]
```diff
@@ -4,7 +4,13 @@ set -xe
 SRC_DIR="${SRC_DIR:-$PWD}"
 DOCKER="${DOCKER:-docker}"
 
-TAG=$(git describe --tags --dirty)
+DOCKER_RUN_ARGS=${DOCKER_RUN_ARGS:-'-ti'}
+
+GO_IMAGE="${GO_IMAGE:-golang}"
+GO_VERSION="${GO_VERSION:-1.18-alpine}"
```
[the remark this answers is not in the document]
Comment (yashsingh74) [reply scripts/release.sh]: done

**Comment by yashsingh74** [comment]
@squeed Please review the PR.

**Pull request closed without merging**